src/main/kotlin/no/nav/syfo/identhendelse/IdenthendelseService.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.syfo.identhendelse
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.dialogmotekandidat.database.getDialogmotekandidatEndringListForPerson
+import no.nav.syfo.dialogmotekandidat.database.getDialogmotekandidatStoppunktList
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.identhendelse.database.*
+import no.nav.syfo.identhendelse.kafka.COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES
+import no.nav.syfo.identhendelse.kafka.KafkaIdenthendelseDTO
+import no.nav.syfo.unntak.database.getUnntakList
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class IdenthendelseService(
+    private val database: DatabaseInterface,
+    private val pdlClient: PdlClient,
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(IdenthendelseService::class.java)
+
+    fun handleIdenthendelse(identhendelse: KafkaIdenthendelseDTO) {
+        if (identhendelse.folkeregisterIdenter.size > 1) {
+            val activeIdent = identhendelse.getActivePersonident()
+            if (activeIdent != null) {
+                val inactiveIdenter = identhendelse.getInactivePersonidenter()
+                val numberOfUpdatedIdenter = updateAllTables(activeIdent, inactiveIdenter)
+                log.info("Identhendelse: Updated $numberOfUpdatedIdenter personoppgaver based on Identhendelse from PDL")
+                COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES.increment(numberOfUpdatedIdenter.toDouble())
+            } else {
+                log.warn("Mangler gyldig ident fra PDL")
+            }
+        }
+    }
+
+    private fun updateAllTables(activeIdent: PersonIdentNumber, inactiveIdenter: List<PersonIdentNumber>): Int {
+        var numberOfUpdatedIdenter = 0
+        numberOfUpdatedIdenter += updateTableWhereOldIdent(
```

Comment:
Er antakelig en noe mer elegant måte å gjøre dette på. Feks lage en liste med 4 par, og så loope igjennom den og kalle updateTableWhereOldIdent for hvert par.

Reply:
Ja, tenkte litt på det, vet ikke hva som blir "penest". Nå har jeg endra litt på arkitekturen også (fjernet `updateTableWhereOldIdent`)